tournamentcontrol/competition/media.py

```diff
@@ -0,0 +1,99 @@
+"""
+Custom media upload classes for YouTube thumbnail management.
+
+This module provides database-sourced media upload classes that are compatible
+with the Google API client library, similar to MediaFileUpload but reading
+data from Django model fields instead of files.
+"""
+
+import io
+from googleapiclient.http import HttpRequest, MediaUpload
+
+
+class MediaDatabaseUpload(MediaUpload):
+    """
+    A MediaUpload subclass that uploads from database-stored binary data.
+    
+    This class mimics the behavior of MediaFileUpload but sources its data
+    from a Django model's BinaryField instead of a file on disk.
+    
+    Args:
+        data (bytes): The binary data to upload
+        mimetype (str): The MIME type of the data
+        chunksize (int): Size of chunks to use for resumable uploads
+        resumable (bool): Whether to make the upload resumable
+    """
+    
+    def __init__(self, data, mimetype, chunksize=1024*1024, resumable=False):
+        super().__init__()
+        self._data = data
+        self._mimetype = mimetype
+        self._chunksize = chunksize
+        self._resumable = resumable
+        self._stream = io.BytesIO(data)
+        
+    def chunksize(self):
+        """Return the upload chunksize."""
+        return self._chunksize
+        
+    def mimetype(self):
+        """Return the mimetype of the upload."""
+        return self._mimetype
+        
+    def size(self):
+        """Return the size of the upload."""
+        return len(self._data)
+        
+    def resumable(self):
+        """Return True if this is a resumable upload."""
+        return self._resumable
+        
+    def getbytes(self, begin, end):
+        """Get bytes from the upload data."""
+        self._stream.seek(begin)
+        return self._stream.read(end - begin)
+        
+    def has_stream(self):
+        """Return True if the upload has an associated stream."""
+        return True
+        
+    def stream(self):
+        """Return the upload stream."""
+        return self._stream
+        
+    def to_json(self):
+        """Convert the upload to JSON representation."""
+        return {
+            'mimetype': self._mimetype,
+            'size': self.size(),
+            'resumable': self._resumable,
+        }
+        
+    @classmethod
+    def from_model_field(cls, obj, image_field_name='thumbnail_image', 
```

Comment:
Rather than `from_model_field`, it's reasonable to expect this to just receive bytes, which decouples it from how and where the bytes are stored.

We have a convenience on the model definition, so this can be entirely agnostic. Also better for unit tests.

So change this to take just bytes and rename accordingly, then move the way the bytes are obtained to somewhere else.